src/sagemaker/model.py

```diff
@@ -2393,3 +2394,44 @@ def add_inference_specification(
         )
 
         sagemaker_session.sagemaker_client.update_model_package(**model_package_update_args)
+
+    def update_model_card(self, model_card: ModelCard | ModelPackageModelCard):
+        """Updates Created model card content which created with model package
+
+        Args:
+            model_card (ModelCard | ModelPackageModelCard): Updated Model Card content
+        """
+
+        sagemaker_session = self.sagemaker_session or sagemaker.Session()
+        desc_model_package = sagemaker_session.sagemaker_client.describe_model_package(
+            ModelPackageName=self.model_package_arn
+        )
+        update_model_card_req = model_card._create_request_args()
+        if update_model_card_req["ModelCardStatus"] is not None:
+            if (
+                desc_model_package["ModelCard"]["ModelCardStatus"]
+                == update_model_card_req["ModelCardStatus"]
+            ):
+                del update_model_card_req["ModelCardStatus"]
+
+        if update_model_card_req.get("ModelCardName") is not None:
+            del update_model_card_req["ModelCardName"]
+        if update_model_card_req.get("Content") is not None:
+            previous_content_hash = _hash_content_str(
+                desc_model_package["ModelCard"]["ModelCardContent"]
+            )
+            current_content_hash = _hash_content_str(update_model_card_req["Content"])
+            if (
+                previous_content_hash == current_content_hash
+                or update_model_card_req.get("Content") == "{}"
+                or update_model_card_req.get("Content") == "null"
+            ):
+                del update_model_card_req["Content"]
```

Comment:
why we doing hashing check

Reply:
validating any updates made on MC content, so that we can remove it from the request to fix from sending both status and content for update which results in error says we cannot update both.